src/uwtools/config/jinja2.py

```diff
@@ -127,7 +127,7 @@ def dereference(val: _ConfigVal, context: dict, local: Optional[dict] = None) ->
     """
     rendered: _ConfigVal = val  # fall-back value
     if isinstance(val, dict):
-        return {k: dereference(v, context, local=val) for k, v in val.items()}
+        return {dereference(k, context): dereference(v, context, local=val) for k, v in val.items()}
```

Comment:
Keys in UW YAML may also contain e.g. `cycle` references that need to be dereferenced, so that's done here now. A use case is the keys in the FV3 driver's `files_to_copy` and `files_to_link` entries, which represent filenames relative to the FV3 run directory and which may need to be cycle-specific.